tasks/docker.rb

```diff
@@ -80,19 +80,25 @@ def fix_ssh(distro, version, container)
 
 def get_image_os_release_facts(image)
   os_release_facts = {}
-  os_release = run_local_command("docker run --rm #{image} cat /etc/os-release")
-  # The or-release file is a newline-separated list of environment-like
-  # shell-compatible variable assignments.
-  re = '^(.+)=(.+)'
-  os_release.each_line do |line|
-    line = line.strip || line
-    next unless !line.nil? && !line.empty?
+  begin
+    os_release = run_local_command("docker run --rm #{image} cat /etc/os-release")
+    # The or-release file is a newline-separated list of environment-like
+    # shell-compatible variable assignments.
+    re = '^(.+)=(.+)'
+    os_release.each_line do |line|
+      line = line.strip || line
+      next unless !line.nil? && !line.empty?
 
-    _, key, value = line.match(re).to_a
-    # The values seems to be quoted most of the time, however debian only quotes
-    # some of the values :/.  Parse it, as if it was a JSON string.
-    value = JSON.parse(value) unless value[0] != '"'
-    os_release_facts[key] = value
+      _, key, value = line.match(re).to_a
+      # The values seems to be quoted most of the time, however debian only quotes
+      # some of the values :/.  Parse it, as if it was a JSON string.
+      value = JSON.parse(value) unless value[0] != '"'
+      os_release_facts[key] = value
+    end
+  rescue
+    id, version_version = image.split(':')
+    os_release_facts['ID'] = id
```

Comment:
wouldn't this set ID to `litmusimage/centos` instead of `centos` and the rest of the machinery is expecting the latter?

Reply:
Have added `id = id.sub(%r{/}, '_')`.
I think this is how it's supposed to go. There was no code in the original method to remove the litmusimage part

Reply:
ok